app/models/ems_refresh.rb

```diff
@@ -171,7 +171,13 @@ def self.queue_merge(targets, ems, create_task = false)
 
     # Items will be naturally serialized since there is a dedicated worker.
     MiqQueue.put_or_update(queue_options) do |msg, item|
-      targets = msg.nil? ? targets : (msg.args[0] | targets)
+      # Handle merging targets for old queue items that still have in args[0]
+      unless msg.args[0].nil?
+        msg.data.concat(msg.args.shift)
+        msg.args = []
+      end
```

Comment:
@Fryguy this works around merging targets into existing queue items that still had their targets in args, do you think this is needed?